catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/HistoryMetacardImpl.java

```diff
@@ -0,0 +1,248 @@
+     *
+     * @since DDF-2.9.0
+     */
+    public static final String ID_HISTORY = "history-id";
+
+    public static final String TAGS_HISTORY = "history-tags";
+
+    private static MetacardType versionHistoryMetacard;
+
+    public static final String METACARD_TYPE_NAME = "ddf.history.metacard";
+
+    static {
+        HashSet<AttributeDescriptor> historyDescriptors =
+                new HashSet<>(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
+        historyDescriptors.add(new AttributeDescriptorImpl(STATE,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(EDITED_BY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(VERSIONED,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.DATE_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(ID_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(TAGS_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                true /* multivalued */,
+                BasicTypes.STRING_TYPE));
+
+        versionHistoryMetacard = new MetacardTypeImpl(METACARD_TYPE_NAME, historyDescriptors);
+    }
+
+    /**
+     * Will convert the given {@link Metacard} to a {@link HistoryMetacardImpl} by cloning
+     * it and adding the current subject, time, and a random UUID. Cannot take a
+     * {@link HistoryMetacardImpl} as the sourceMetacard.
+     *
+     * @param sourceMetacard Metacard to clone and create a history item from
+     * @param action         Which action was done to modify the metacard
+     * @throws IllegalArgumentException
+     */
+    public HistoryMetacardImpl(Metacard sourceMetacard, Action action) {
+        super(sourceMetacard, versionHistoryMetacard);
+        if (sourceMetacard instanceof HistoryMetacardImpl) {
+            throw new IllegalArgumentException(
+                    "Cannot create a history item from a history metacard.");
+        }
+
+        this.setState(action);
+        this.setIdHistory(sourceMetacard.getId());
+        this.setTagsHistory(sourceMetacard.getTags());
+
+        String editedBy = SubjectUtils.getEmailAddress(SecurityUtils.getSubject());
```

Comment:
I think you should probably be passing the Subject in via the constructor instead of trying to snag it here. Trying to pull the subject off of the thread local at this point really constrains what you can do with this impl.